src/app/pages/errata-summary/hero/hero.js

```diff
@@ -34,29 +29,80 @@ function useBookInfo(book) {
     return info;
 }
 
-function useErrataHoverHtml(slug) {
-    const [html, setHtml] = useState('<p>...loading...</p>');
+const middle = '-135';
+
+function shiftIntoView(ref, leftOffset, setLeftOffset) {
+    const {left, right} = ref.current.getBoundingClientRect();
+    const {right: pageRight} = ref.current.closest('.page').getBoundingClientRect();
+    const overRight = right - pageRight - 15;
+    const overLeft = 15 - left;
+    const rightWants = Math.min(middle, leftOffset - overRight);
+    const leftWants = Math.max(rightWants, leftOffset + overLeft);
+
+    setLeftOffset(leftWants);
+}
+
+function usePopTipStyle(isOpen) {
+    const ref = React.useRef();
+    const [leftOffset, setLeftOffset] = React.useState(middle);
 
     useEffect(() => {
-        async function fetchData() {
-            if (!slug) {
-                return;
-            }
-            const [bookData, errataPageData] = await Promise.all([cmsFetch(slug), cmsFetch('pages/errata')]);
-            const k = correctionScheduleKeyFromBookState[bookData.book_state];
-            const errataMessage = errataPageData[k] || `Book in unknown state: ${bookData.book_state}`;
+        shiftIntoView(ref, leftOffset, setLeftOffset);
+    }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    return {ref, style: {left: leftOffset}};
+}
+
+function PopTip({html, isOpen}) {
+    const {ref, style} = usePopTipStyle(isOpen);
+
+    return (
+        <div className="poptip-attachment">
+            <div className="poptip" ref={ref} style={style} role="tooltip" aria-hidden={!isOpen}>
+                <div className="title-bar">
+                    More information
+                </div>
+                <RawHTML className="body" html={html} />
+            </div>
+        </div>
+    );
+}
```

Comment:
is this the wrapper element?  if so should it have `aria-alert` set?

Reply:
This is not an alert. I have set role to `tooltip`.

Reply:
seems like role="tooltip" does the same though, so never mind!